### PR TITLE
AI notification update

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

This PR creates a stripped down notify_ghost system for the AI.
In layman's terms, this means the AI gets on screen notifications for things instead of wading through chat. The AI can then click the notifications to go to the triggering event. 

Right now the AI receives a notification for the following events;

When the Alamo is locked by xeno activity
When the tadpole is destroyed
When a disk is generated
When the nuke is activated
When a CAS laser is activated
When a OB targeting laser is activated
When a railgun targeting laser is activated
When a fresh clone becomes active
When they are summoned via holopad

The system is easily added to, and it can be expanded upon as needed. With the exception of the OB laser, clicking any of the alerts will take you to the site of the event.

## Why It's Good For The Game

For what is theoretically the embodiment of the ship's intelligence the AI really doesn't have that much info about operations. It can't tell when much of anything happens, aside from basic statistics. With this PR the AI should be able to stay better informed about the state of the operation, and act as auxiliary command as needed.

## Changelog
:cl:
qol: The AI now receives alerts about important events.
code: Created a modified version of the ghost notification system for AI use.
/:cl:

### CODE REVIEW

File: code/game/objects/effects/overlays.dm [hunk 1]
```diff
@@ -187,6 +189,13 @@
 	squad = assigned_squad
 	if(squad)
 		squad.squad_laser_targets += src
+	switch(lasertype)
+		if("railgun")
```
Comment (BraveMole) [inline code/game/objects/effects/overlays.dm:193]: Should be defines

File: code/modules/mob/living/silicon/ai/ai_notifications.dm [hunk 2]
```diff
@@ -0,0 +1,85 @@
+
+/mob/living/silicon/ai/proc/notify_ai(mob/living/silicon/receivingai, message, ai_sound = null, atom/source = null, mutable_appearance/alert_overlay = null, action = NOTIFY_AI_ALERT, header = null, notify_volume = 100) //stripped down notify_ghost for AI
+
+	if(ai_sound)
+		SEND_SOUND(receivingai, sound(ai_sound, volume = notify_volume, channel = CHANNEL_NOTIFY))
+	if(!source)
+		return
+
+	var/atom/movable/screen/alert/ai_notify/A = receivingai.throw_alert("[REF(source)]_notify_action", /atom/movable/screen/alert/ai_notify)
+	if(!A)
```
Comment (BraveMole) [inline code/modules/mob/living/silicon/ai/ai_notifications.dm:10]: please, full variable name

File: code/modules/mob/living/silicon/ai/ai_notifications.dm [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+	A.add_overlay(alert_overlay)
+
+///Receive fire support laser notifications
+/mob/living/silicon/ai/proc/receive_laser_ob(datum/source, obj/effect/overlay/temp/laser_target/OB/incoming_laser)
+	SIGNAL_HANDLER
+	to_chat(src, span_notice("Orbital Bombardment laser detected. Target: [AREACOORD_NO_Z(incoming_laser)]"))
+	notify_ai(src, "<b> An Orbital Bombardment laser</b> has been detected at [AREACOORD_NO_Z(incoming_laser)]!", ai_sound = 'sound/effects/obalarm.ogg', source = incoming_laser, action = NOTIFY_AI_BOMBARD, notify_volume = 15)
+
+/mob/living/silicon/ai/proc/receive_laser_cas(datum/source, obj/effect/overlay/temp/laser_target/cas/incoming_laser)
```
Comment (BraveMole) [inline code/modules/mob/living/silicon/ai/ai_notifications.dm:41]: autodoc on everything

**Review by BraveMole — CHANGES_REQUESTED** [review]
The ob solution cannot pass, you probably want to cut it so the rest can be merged.

File: code/_onclick/hud/ai.dm [hunk 4]
```diff
@@ -150,3 +150,35 @@
 	using = new /atom/movable/screen/ai/add_multicam()
 	using.screen_loc = ui_ai_add_multicam
 	static_inventory += using
+
+/atom/movable/screen/alert/ai_notify
+	name = "Notification"
+	desc = "A new notification. You can enter it."
+	icon_state = "template"
+	timeout = 15 SECONDS
+	var/atom/target = null
```
[the remark this answers is not in the document]
Comment (RipGrayson) [reply code/_onclick/hud/ai.dm:159]: In this case, /mob/living/silicon/ai/proc/notify_ai sets the target for /atom/movable/screen/alert/ai_notify when called

**Review by Lumipharon — APPROVED** [review]
